map/set-max-extent/src/main/java/com.esri.samples.set_max_extent/SetMaxExtentSample.java

```diff
@@ -0,0 +1,142 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.set_max_extent;
+
+import javafx.application.Application;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.control.ToggleGroup;
+import javafx.scene.layout.*;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.symbology.*;
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.BasemapStyle;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class SetMaxExtentSample extends Application {
+
+    private MapView mapView;
+
+
+    @Override
+    public void start(Stage stage) {
+
+        try {
+            // create stack pane and application scene
+            var stackPane = new StackPane();
+            var scene = new Scene(stackPane);
+
+            // set title, size, and add scene to stage
+            stage.setTitle("Set Max Extent Sample");
+            stage.setWidth(800);
+            stage.setHeight(700);
+            stage.setScene(scene);
+            stage.show();
+
+            // authentication with an API key or named user is required to access basemaps and other location services
+            String yourAPIKey = System.getProperty("apiKey");
+            ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+            // create a map with the streets basemap style
+            var map = new ArcGISMap(BasemapStyle.ARCGIS_STREETS);
+
+            // create a map view and set the map to it
+            mapView = new MapView();
+            mapView.setMap(map);
+
+            // create a max extent envelope with points on Colorado's northwest and southeast corners
+            var coloradoNW = new Point(-12139393.2109, 5012444.0468, SpatialReferences.getWebMercator());
+            var coloradoSE = new Point(-11359277.5124, 4438148.7816, SpatialReferences.getWebMercator());
+            var extentEnvelope = new Envelope(coloradoNW, coloradoSE);
+
+            // set the map's max extent to the max extent envelope
+            map.setMaxExtent(extentEnvelope);
```

Comment:
with the Java Runtime API, it's best practice to listen for the map loading to finish, and if it loaded successfully, to then carry out map related tasks. We can move this into an addDoneLoadingListener block as follows (with a bonus setting the viewpoint of the map view: another common workflow). It also allows us to display a message if the map fails to load: if the map fails to load then there's no need to set its max extent! 

```suggestion
      // listen for the map to finish loading, and check it has loaded
      map.addDoneLoadingListener(() -> {
        if (map.getLoadStatus() == LoadStatus.LOADED) {
          // constrain the display of the map to the state borders of Colorado by setting the map's max extent to the envelope
          map.setMaxExtent(envelope);
          // set the map view's viewpoint with the envelope
          mapView.setViewpoint(new Viewpoint(envelope));
        } else if (map.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
        }
      });
```